bitvm/src/bn254/fp254impl.rs

```diff
@@ -686,6 +686,46 @@ pub trait Fp254Impl {
         (script, hints)
     }
 
+    #[allow(clippy::too_many_arguments)]
+    fn hinted_mul_lc2_w4(
+        a_depth: u32,
+        a: ark_bn254::Fq,
+        b_depth: u32,
+        b: ark_bn254::Fq,
+        c_depth: u32,
+        c: ark_bn254::Fq,
+        d_depth: u32,
+        d: ark_bn254::Fq,
+    ) -> (Script, Vec<Hint>) {
+        assert!(a_depth > b_depth && b_depth > c_depth && c_depth > d_depth);
+
+        let mut hints = Vec::new();
```

Comment:
Looks like the size is known beforehand, can we make it explicit? Same to the others.

```suggestion
        let mut hints = Vec::new_capacity(1);
```

Reply:
fixed in latest commit. 
I have made the size specification only for changes scoped within this PR.